dropshot/src/server.rs

```diff
@@ -194,20 +198,30 @@ impl<C: ServerContext> HttpServerStarter<C> {
             })
             .transpose()?;
         let handler_waitgroup = WaitGroup::new();
+
+        let router = api.into_router();
+        if let VersionPolicy::Unversioned = version_policy {
+            if router.has_versioned_routes() {
+                return Err(BuildError::UnversionedServerHasVersionedRoutes);
```

Comment:
is this the point of `has_versioned_routes` i.e. is it mostly here to produce an error message if the user has done something aberrant?